File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int pLen = p.length();
        long small = Long.parseLong(p);
        
        for (int i = 0; i <= t.length() - pLen; i++){
            long big = Long.parseLong(t.substring(i, i + pLen));

            if (big <= small) answer += 1;
        }

        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    len_t = len(t)
    len_p = len(p)
    
    for i in range(len_t-len_p+1):
        if int(t[i:i+len_p]) <= int(p):
            answer += 1
    
    return answer